jython_cli.java

```diff
@@ -0,0 +1,212 @@
+                    if (line.startsWith("# /// jbang")) {
+                        found = true;
+                    }
+                    else if (line.startsWith("# ///")) {
+                        found = false;
+                        break;
+                    } else if (line.startsWith("# ")) {
+                        if (found) {
+                            if (tomlText.length() == 0) {
+                                tomlText.append(line.substring(2));
+                            } else {
+                                tomlText.append(ls + line.substring(2));
+                            }
+                        }
+                    }
+                }
+            }
+            TomlParseResult tpr = Toml.parse(tomlText.toString());
+            // [python-jvm]
+            TomlTable pythonjvmTable = tpr.getTable("jython-cli");
+            if (pythonjvmTable != null) {
+                if (pythonjvmTable.isBoolean("debug")) {
+                    debug = pythonjvmTable.getBoolean("debug");
+                }
+            }
+            if (debug) {
+                System.out.println("[debug] python-jvm init " + javaFilename + " from " + scriptFilename);
+            }
+            if (debug) {
+                System.out.println("");
+                System.out.println("[ -----------------jbang-config-begin-------------------- ]");
+                System.out.println("");
+                System.out.println(tpr.toToml());
+                System.out.println("[ -----------------jbang-config-end---------------------- ]");
+                System.out.println("");
+            }
+            if (tpr.isString("requires-jython")) {
+                jythonVersion = tpr.getString("requires-jython");
+            }
+            if (tpr.isString("requires-java")) {
+                javaVersion = tpr.getString("requires-java");
+            }
+            // dependencies
+            for (Object e : tpr.getArrayOrEmpty("dependencies").toList()) {
+                String dep = (String) e;
+                deps.add(dep);
+            }
+            // [java]
+            TomlTable javaTable = tpr.getTable("java");
+            if (javaTable != null) {
+                String runtimeOptions = javaTable.getString("runtime-options");
+                if (runtimeOptions != null) {
+                    javaRuntimeOptions = runtimeOptions;
+                }
+            }
+            // [jbang]
+            TomlTable jbangTable = tpr.getTable("jbang");
+            if (jbangTable != null) {
+                Boolean integrations = jbangTable.getBoolean("integrations");
+                if (integrations != null && integrations.equals(Boolean.FALSE)) {
+                    jbangIntegrations = "false";
+                }
+            }
+        }
+
+        String dep = "org.python:jython-standalone:" + jythonVersion;
+        deps.add(dep);
+
+        byte[] data = Files.readAllBytes(Paths.get(scriptFilename));
```

Comment:
Well this is a surprise ... so here we write a new Java program that is specific to the original Python script, and contains some stuff from the JBang directives in the Python. When run, with JBang, it invokes `PythonInterpreter`.

This seems very indirect. I think I was expecting argument to this tool either to go directly to `o.p.util.jython.main()` or for Java to execute the JAR with those.

I realise this surprise comes because nothing so far has explained what this program actually does, and how it goes about it.

Reply:
@jeff5 , I thought about your comment and realized the added complexity was really a left-over from code I wrote to embed a Jython script as a reporting tool in a larger Java/Kotlin application and that this functionality is not needed this scenario. I removed the script caching code and clean-up the Java code. It now ends up  just calling jython.main(args). I'll explain the program logic in a follow up comment.